modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/store/TransactionData.java

```diff
@@ -103,4 +109,14 @@ public boolean isEmpty() {
     public int size() {
         return detachOperationsCount + enqueueMessages.size() + deleteMessageIdList.size();
     }
+
+    public void completePostCommitActions() {
+        for (Map.Entry<String, QueueDetachEventList> entry: detachMessageMap.entrySet()) {
+            List<Runnable> postTransactionActions = entry.getValue().getPostCommitActions();
+            for (Runnable action: postTransactionActions) {
+                action.run();
```

Comment:
What happens if we get an Runtime exception?

Reply:
ATM it won't throw any Runtime exceptions. Anyway I'm gonna remove this when implementing the transaction frame.